site/pages/api/createDisruption.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    MiscellaneousReason,
+    PersonnelReason,
+    EnvironmentReason,
+    EquipmentReason,
+} from "@create-disruptions-data/shared-ts/siriTypes";
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+    COOKIES_DISRUPTION_INFO,
+    COOKIES_DISRUPTION_ERRORS,
+    TEN_SECONDS_IN_MILLISECONDS,
+    CREATE_DISRUPTION_PAGE_PATH,
+    ERROR_PATH,
+} from "../../constants/index";
+import { ErrorInfo } from "../../interfaces";
+import { setCookieOnResponseObject } from "../../utils/apiUtils";
+import {
+    checkReferrer,
+    validateDisruptionType,
+    validateSummary,
+    validateDescription,
+    validateAssociatedLink,
+    validateDisruptionReasons,
+    validateDateTimeSection,
+    getDateTime,
+} from "../../utils/apiUtils/createDisruptionValidations";
+import { redirectTo } from "../../utils/index";
+import { PageInputs } from "../create-disruption";
+
+const createDisruption = (req: NextApiRequest, res: NextApiResponse): void => {
+    const errors: ErrorInfo[] = [];
+
+    checkReferrer(req.headers.referer, CREATE_DISRUPTION_PAGE_PATH, ERROR_PATH, res);
+
+    const formFields: PageInputs = req.body as PageInputs;
+
+    const disruptionType: "planned" | "unplanned" | undefined = formFields.typeOfDisruption;
+
+    validateDisruptionType(disruptionType, errors, "some-error-id");
+
+    const summary = formFields.summary;
+
+    validateSummary(summary, errors, "some-error-id");
+
+    const description = formFields.description;
+
+    validateDescription(description, errors, "some-error-id");
+
+    const associatedLink = formFields["associated-link"];
+
+    if (associatedLink) {
+        validateAssociatedLink(associatedLink, errors, "some-error-id");
+    }
+
+    const disruptionReason: MiscellaneousReason | PersonnelReason | EnvironmentReason | EquipmentReason | "" =
+        formFields["disruption-reason"] || "";
+
+    validateDisruptionReasons(disruptionReason, errors, "some-error-id");
+
+    const disruptionStartDate: Date | null = formFields["disruption-start-date"];
+    const disruptionStartTime: string = formFields["disruption-start-time"];
+    const disruptionEndDate: Date | null = formFields["disruption-end-date"];
+    const disruptionEndTime: string = formFields["disruption-end-time"];
+    const disruptionIsNoEndDateTime: string | undefined = formFields.disruptionIsNoEndDateTime;
+
+    validateDateTimeSection(
+        disruptionStartDate,
+        disruptionStartTime || "",
+        disruptionEndDate,
+        disruptionEndTime || "",
+        disruptionIsNoEndDateTime,
+        errors,
+        "some-error-id",
```

Comment:
If this is just an un-needed value, can we not just pass "" or make it optional?

Reply:
It is a placeholder to indicate that an error id should be passed. As the value is not currently used and will be updated later, using a text or empty string would be the same.